src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Mvc/InjectTargetExtension.cs

```diff
@@ -42,7 +42,7 @@ public void WriteInjectProperty(CodeRenderingContext context, InjectIntermediate
         else if (!node.IsMalformed)
         {
             var property = $"public {node.TypeName} {node.MemberName} {{ get; private set; }}";
-            if (!context.Options.SuppressNullabilityEnforcement)
+            if (considerNullabilityEnforcement && !context.Options.SuppressNullabilityEnforcement)
```

Comment:
Can't we set `context.Options.SuppressNullabilityEnforcement` in the tests where we need it instead of introducing a new test-only parameter?

Reply:
No, this isn't about testing, but actual implementation. We don't ever want to write out the nullability info for the older MVC versions. There probably isn't any harm in doing so, but I wanted to make sure the emitted code was exactly the same as part of this PR.